modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/DeleteOneTree.pm

```diff
@@ -47,8 +47,17 @@ sub write_output {
         $tree->preload;
         $gene_tree_adaptor->delete_tree($tree);
         $tree->release_tree;
-    } );
+    }, 1, 2 );  # Retry once and wait 2 seconds between retries
```

Comment:
If the `sub` fails, the transaction code will retry it if `$retry` is set, or throw an error otherwise. You're working on the premise that in e101, something went wrong but no error was raised. Based on that, I don't see how setting `$retry` to 1 will make it try again this time.

Reply:
I was basing it a little on something that happened to me when I was trying to copy/replace the trees from the previous release and delete the unstructured trees with `delete_tree()`. I wasn't sure what triggers the `$retry`. In my experiments, when the trees were very large `delete_tree()` struggled, and the processes timed out before they were reported as such, I wondered if it was a catch-up thing or something to do with rollback and perhaps pausing for a couple of seconds might help. What you say makes sense though. Remove this bit of possibly useless overkill?

Reply:
Since we don't understand how the trees were not deleted, I shouldn't try to be logical :) Let's keep it and if we're lucky we won't face the issue again